app/js/lib/foxbox/recipes.js

```diff
@@ -0,0 +1,300 @@
+'use strict';
+
+// @todo Proxy recipes in the db and emit events on change (see services).
+
+// Private members.
+const p = {
+  // Private properties.
+  settings: Symbol('settings'),
+  net: Symbol('net')
+};
+
+export default class Recipe {
+  constructor(props) {
+    this[p.settings] = props.settings;
+    this[p.net] = props.net;
+  }
+
+  /**
+   * Returns a promise resolving to a list of recipes.
+   *
+   * @return {Promise}
+   */
+  getAll() {
+    return this[p.net].fetchJSON(
+      `${this[p.net].origin}/api/v${this[p.settings].apiVersion}/services`)
+      // Get all recipes.
+      .then(services => services.filter(service =>
+        service.adapter === 'thinkerbell-adapter' &&
+        service.id !== 'thinkerbell-root-service'
+      ))
+      // Fetch their respective enabled status.
+      .then(recipes => {
+        const promises = recipes.map(recipe => {
+          const payload = [
+            { id: `${recipe.id}/get_enabled` }
+          ];
+
+          return this[p.net].fetchJSON(
+            `${this[p.net].origin}/api/v${this[p.settings].apiVersion}/` +
+            'channels/get',
+            'PUT',
+            payload);
+        });
+
+        return Promise.all(promises)
+          .then(servicesEnabled => {
+            return servicesEnabled;
+          })
+          // Map all the recipes to a more user-friendly format.
+          .then(servicesEnabled => recipes.map((recipe, id) => ({
+            id: recipe.id,
+            label: recipe.id,
+            enabled: servicesEnabled[id][`${recipe.id}/get_enabled`]
+              .OnOff === 'On'
+          })));
+      });
+  }
+
+  getServicesWithGetters() {
```

Comment:
Actually that's a significant refactoring that will impact the view too.
I'll do it later.